src/org/omegat/core/machinetranslators/YandexCloudTranslate.java

```diff
@@ -230,4 +264,39 @@ private String getIAMToken(final String oAuthToken) {
         }
         return cachedIAMToken;
     }
+
+    private String getGlossaryConfigPart(SourceTextEntry ste) {
+
+        if (!Preferences.isPreference(PROPERTY_USE_GLOSSARY)) {
+            return EMPTY_STRING;
+        }
+
+        List<GlossaryEntry> glossaryHits = Core.getGlossaryManager().searchSourceMatches(ste);
```

Comment:
I don’t really like MT providers being coupled to the core. Avoiding this is why I suggested the entry point should just accept a list of glossary terms instead of the `SourceTextEntry`.

Reply:
Wouldn't passing a `GlossaryManager` be a compromise choice here? Passing glossary terms would take a search on each MT call and most of MT engines don't use those terms.